source/containers.tex

```diff
@@ -9296,14 +9296,15 @@
     typename iterator_traits<InputIterator>::value_type;                // \expos
 template<class InputIterator>
   using @\placeholder{iter-key-type}@ = remove_const_t<
-    typename iterator_traits<InputIterator>::value_type::first_type>;   // \expos
+    tuple_element_t<0, @\exposid{iter-value-type}@<InputIterator>>>;                // \expos
 template<class InputIterator>
   using @\placeholder{iter-mapped-type}@ =
-    typename iterator_traits<InputIterator>::value_type::second_type;   // \expos
+    tuple_element_t<1, @\exposid{iter-value-type}@<InputIterator>>;                 // \expos
 template<class InputIterator>
   using @\placeholder{iter-to-alloc-type}@ = pair<
-    add_const_t<typename iterator_traits<InputIterator>::value_type::first_type>,
-    typename iterator_traits<InputIterator>::value_type::second_type>;  // \expos
+    add_const_t<tuple_element_t<0, @\exposid{iter-value-type}@<InputIterator>>>,
+    tuple_element_t<1, @\exposid{iter-value-type}@<InputIterator>>>;                // \expos
```

Comment:
Why didn't these reuse _`iter-mapped-type`_?

Reply:
Not an editorial issue or part of this paper - please send to lwgchair. Ok to resolve here?